pystiche_papers/sanakoyeu_et_al_2018/_modules.py

```diff
@@ -184,3 +190,58 @@ def residual_block(channels: int) -> ResidualBlock:
     )
 
     return ResidualBlock(residual)
+
+
+def pairwise(iterable: Iterable) -> Iterator[Tuple[Any, Any]]:
+    a, b = itertools.tee(iterable)
+    next(b, None)
+    return zip(a, b)
+
+
+def discriminator_encoder_modules(
+    in_channels: int = 3, inplace: bool = True
+) -> List[Tuple[str, nn.Sequential]]:
+    conv_block = partial(
+        ConvBlock, kernel_size=5, stride=2, padding="same", act="lrelu", inplace=inplace
+    )
+
+    channels = (in_channels, 128, 128, 256, 512, 512, 1024, 1024)
+    return [
+        (f"scale_{idx}", conv_block(in_channels, out_channels))
+        for idx, (in_channels, out_channels) in enumerate(pairwise(channels))
+    ]
+
+
+class DiscriminatorMultiLayerEncoder(MultiLayerEncoder):
+    r"""Discriminator from :cite:`SKL+2018`.
+
+    Args:
+        in_channels: Number of channels in the input. Defaults to ``3``.
+    """
+
+    def __init__(self, in_channels: int = 3) -> None:
+        super().__init__(discriminator_encoder_modules(in_channels=in_channels))
+
+
+def prediction_module(
```

Comment:
IIRC this is part of the `style_loss`, correct? If that is the case, please revert his for now and include this in the PR for the `style_loss`.